modules/scriptworker/manifests/chain_of_trust.pp

```diff
@@ -32,47 +16,9 @@
   }
 
   file {
-    # $username's gpg homedir: for git commit signature verification
-    "/home/${username}/.gnupg":
-        ensure      => directory;
-    # these are the pubkeys that can sign git commits
-    $git_pubkey_dir:
-        ensure       => directory,
-        source       => 'puppet:///modules/scriptworker/git_pubkeys',
-        recurse      => true,
-        recurselimit => 1,
-        purge        => true,
-        require      => Python3::Virtualenv[$basedir];
-    # cron jobs to poll git + rebuild gpg homedirs
     '/etc/cron.d/scriptworker':
-        content     => template('scriptworker/scriptworker.cron.erb');
-    # Notify rebuild_gpg_homedirs if the pubkey dir changes
-    "${basedir}/.git-pubkey-dir-checksum":
-        notify  => Exec['rebuild_gpg_homedirs'];
-    "/home/${username}/pubkey":
-        mode      => '0644',
-        content   => $gpg_pubkey,
-        show_diff => true;
-    "/home/${username}/privkey":
-        content => $gpg_privkey;
+        ensure     => absent;
     "/home/${username}/ed25519_privkey":
         content => $ed25519_privkey;
   }
-
-  exec {
-      # create gpg homedirs on change
-      'rebuild_gpg_homedirs':
-          require   => [Python3::Virtualenv[$basedir],
-                      Git::Repo["scriptworker-${git_key_repo_dir}"],
-                      File["${basedir}/scriptworker.yaml"]],
-          command   => "${basedir}/bin/rebuild_gpg_homedirs ${basedir}/scriptworker.yaml",
-          subscribe => File[$git_pubkey_dir],
-          user      => $username;
-      # Create checksum file of git pubkeys
-      "${basedir}/.git-pubkey-dir-checksum":
-          require => File[$git_pubkey_dir],
-          path    => '/usr/local/bin/:/bin:/usr/sbin:/usr/bin',
-          user    => $username,
-          command => "find ${git_pubkey_dir} -type f | xargs md5sum | sort > ${basedir}/.git-pubkey-dir-checksum";
-  }
```

Comment:
I think all the files referenced here (both directly, and via exec) probably want to be marked as absent?

Reply:
I thought about that. I leaned towards keeping them, and getting rid of the last traces by migrating to docker/GCP. Because they're no longer valid keys according to scriptworker, I'm less concerned here. I can change the patch to specify `absent`, but I'll have to keep the configs in place to do so.

Reply:
It *looks* like all the files are fixed relative to the basedir, so you could hard-code the things to remove? I don't think anything actually uses the flexibility to specify them individually.

Reply:
Smart. https://grokbase.com/t/gg/puppet-users/1446kqrt5g/how-to-remove-a-directory#201404083akctk6dr7pmqf5cndxfste3ky worked!